### PR TITLE
Cleanup, and little improvement for aggressive trimming

### PR DESCRIPTION
**Pull request by radical**
- The `ConfigureTrimming` target adds xunit assemblies explicitly, and
  sets `TriggerRootAssembly` too.

- Looking at the files being linked, some issues noticed:
    - Multiple copies of xunit assemblies, from various locations are
      included. Eg. from `System.Buffer.Tests`, `WasmTestRunner`, and
      from the nuget
    - `WasmTestRunner` is explicitly added as `TriggerRootAssembly`, but
      it is also passed as an arg for linking, `-p link ...`
    - `xunit.runner.json` is incorrectly added to the files to be linked

- Instead:
    - all the files to be published are already in
      `ResolvedFileToPublish` after `AddTestRunnersToPublishedFiles` target
    - so, we mark everything for linking,
    - *except* the main test assembly
    - Also, we use the test runner as a root

    - This way we are explicitly specifying which assemblies to link,
      and which ones to `copy`

- This reduces the size:
    - 3.1M to 2.6M `artifacts/bin/System.Buffers.Tests/net6.0-Release/browser-wasm/AppBundle/managed/`
    - 46M to 31M `artifacts/bin/System.Buffers.Tests/net6.0-Release/browser-wasm/AppBundle/publish/`

### CODE REVIEW

File: eng/testing/tests.mobile.targets [hunk 1]
```diff
@@ -148,26 +148,22 @@
...
-        <TrimMode>copy</TrimMode>
-      </ManagedAssemblyToLink>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Android'" Include="AndroidTestRunner"/>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'Browser'" Include="WasmTestRunner"/>
-      <TrimmerRootAssembly Condition="'$(TargetOS)' == 'iOS'" Include="AppleTestRunner"/>
-      <TrimmerRootAssembly Include="$(AssemblyName)"/>
+      <!-- Mark all the assemblies for link. We will explicitly mark the non-trimmable ones -->
+      <ResolvedFileToPublish TrimMode="link" />
+
+      <!-- Don't trim the test runner, and main assembly.
```
Comment (marek-safar) [inline eng/testing/tests.mobile.targets:160]: We want to trim the runner just not the test assembly

Comment (radical) [reply eng/testing/tests.mobile.targets:160]: ah ok. We were earlier passing `-a WasmTestRunner` to `illink.dll` - does that not mean that we shouldn't be trimming it?

Comment (radical) [reply eng/testing/tests.mobile.targets:160]: With the changes here we get:

```
                    -a "/Users/radical/dev/r2/artifacts/bin/WasmTestRunner/net6.0-Release/WasmTestRunner.dll"
                     -a "/Users/radical/dev/r2/artifacts/obj/System.Buffers.Tests/net6.0-Release/browser-wasm/System.Buffers.Tests.dll"

                     -reference "/Users/radical/dev/r2/artifacts/bin/WasmTestRunner/net6.0-Release/WasmTestRunner.dll"
                     -p copy "WasmTestRunner"
                     -reference "/Users/radical/dev/r2/artifacts/obj/System.Buffers.Tests/net6.0-Release/browser-wasm/System.Buffers.Tests.dll"
                     -p copy "System.Buffers.Tests"
```

.. instead we want only the test assembly to be `copy`. But what should the `-a` args be?

**Review by mdh1418 — APPROVED** [review]
Thanks! Looks good to me.

File: eng/testing/tests.mobile.targets (hunk 1 above)
Comment (marek-safar) [reply eng/testing/tests.mobile.targets:160]: -a args are correct, you need to tweak `-p copy "WasmTestRunner"` to be `-p link "WasmTestRunner"` though

**Review by steveisok — APPROVED** [review]
Excellent!  Thanks for trimming even more.

**Review by lewing — APPROVED** [review]
👍

**Review by marek-safar — APPROVED** [review]

**Pull request merged**